concordium-std/src/traits.rs

```diff
@@ -404,6 +405,38 @@ pub trait HasLogger {
     }
 }
 
+/// Objects which provide cryptographic utility functions.
+pub trait HasCryptoUtils {
+    /// Verify an ed25519 signature.
+    fn verify_ed25519_signature(
+        &self,
+        public_key: PublicKeyEd25519,
+        signature: SignatureEd25519,
+        message: &[u8],
+    ) -> bool;
+
+    /// Verify an ecdsa signature over secp256k1 with the bitcoin-core
+    /// implementation.
+    ///
+    /// **The message must be a cryptographically safe hash. Otheriwse, the
+    /// signature may leak.**
```

Comment:
I know this used to be part of the comment I put next to the primitive, but please remove it, since this is not entirely accurate.